docs/dev/Lua API.rst

```diff
@@ -3928,6 +3928,211 @@ Examples:
+
+* ``dateTime:year()``
+
+  Returns an integer pair: *year*, *year tick count*.
+
+* ``dateTime:setYear(year)``
+
+  Sets the year. Returns *DateTime*.
+
+* ``dateTime:setYearTick(ticks)``
+
+  Sets year_tick to ticks, normalizing if needed.
+  It is advised to use this function instead of
+  assigning to year_tick directly due to normalization.
+  returns *DateTime*.
+
+* ``dateTime:dayOfYear()``
+
+  Returns an integer pair: *the current day of the year starting from one*, *day tick count*.
+
+* ``dateTime:setDayOfYear(day)``
+
+  Sets the day of year starting from one.
+  returns *DateTime*.
+
+* ``dateTime:dayOfMonth()``
+
+  Returns an integer pair: *the current day of the month starting from one*, *day tick count*.
+
+* ``dateTime:setDayOfMonth(day[, month])``
+
+  Sets the day of month. If month is not given, defaults to current month.
+  returns *DateTime*.
+
+* ``dateTime:dayOfMonthWithSuffix()``
+
+  Returns *day of month, as a string, including its ordinal suffix*.
+
+* ``dateTime:month()``
+
+  Returns *current month, as a number, starting from one*.
+
+* ``dateTime:setMonth(month)``
+
+  Sets the current month starting from one.
+  Returns *DateTime*.
+
+* ``dateTime:nameOfMonth()``
+
+  Returns *name of the current month*.
+
+* ``dateTime:dayOfYearTick()``
+
+  Returns *days from ticks since beginning of current year starting from zero*.
+
+* ``dateTime:dayOfMonthTick()``
+
+  Returns *days from ticks since the beginning of current month starting from zero*.
+
+* ``dateTime:dayTick()``
+
+  Returns *ticks since beginning of the current day starting from zero*.
+
+* ``dateTime:setDayTick(ticks)``
+
+  Sets the current day's tick count.
+
+* ``dateTime:monthOfYearTick()``
+
+  Returns *months from ticks since the beginning of the current year starting from zero*.
+
+* ``dateTime:monthTick()``
+
+  Returns *ticks since the beginning of a month*.
+
+* ``dateTime:daysToTicks(days)``
+
+  Returns *days converted to ticks*.
+
+* ``dateTime:monthsToTicks(months)``
+
+  Returns *months converted to ticks*.
+
+* ``dateTime:addTicks(ticks)``
+
+  Adds *ticks*, normalizing if needed.
+  Returns *DateTime*.
+
+* ``dateTime:addDays(days)``
+
+  Adds *days*, normalizing if needed.
+  Returns *DateTime*.
+
+* ``dateTime:addMonths(months)``
+
+  Adds *months*, normalizing if needed.
+  Returns *DateTime*.
+
+* ``dateTime:addYears(years)``
+
+  Adds *years* to dateTime. Returns *DateTime*.
+
+* ``dateTime:time()``
+
+  Returns three integers: *hours (24 hour time)*, *minutes*, *seconds* of the current day.
+
+* ``dateTime:setTime(hours, minutes, seconds)``
+
+  Sets *hours*, *minutes*, *seconds* of the current day.
+  returns *DateTime*.
+
+* ``dateTime:toDuration()``
+
+  Returns a *Duration object*.
+
+* ``dateTime:normalize()``
+
+  Ensure's year_tick is within bounds of a year, otherwise
+  adjusts year and year_tick accordingly. Note: should only
+  call this function when directly modifying year_tick.
```

Comment:
this *should* not need to be exposed. what reason would callers have to modify `year_tick` directly?

Reply:
Doesn't need to be exposed. Thinking was: 'you shouldn't be assigning to year_tick yourself, but if you insist then here's this function'